lib/unicorn/http_server.rb

```diff
@@ -475,7 +478,10 @@ def murder_lazy_workers
       next_sleep = 0
       logger.error "worker=#{worker.nr} PID:#{wpid} timeout " \
                    "(#{diff}s > #{@timeout}s), killing"
-      before_murder.call(self, worker, wpid) if before_murder
+      if before_murder and not @before_murder_called.include?(wpid)
```

Comment:
We prefer `&& !` instead of `and not` here. Use of the english operators should be restricted to control flow (like the Perl-ism: `do_this or die`).